back/src/bsdasris/resolvers/mutations/updateSimpleBsdasri.ts

```diff
@@ -52,36 +52,38 @@ const updateBsdasri = async ({
   const autocompletedInput = await recipify(sirenifiedInput);
   const flattenedInput = flattenBsdasriInput(autocompletedInput);
 
-  if (inputGrouping && inputGrouping.length > 0 && !isGroupingType) {
-    throw new UserInputError(
-      "Le champ grouping n'est accessible que sur les dasri de groupement."
-    );
-  }
-
   if (inputSynthesizing && inputSynthesizing.length > 0) {
     throw new UserInputError(
       "Le champ synthesizing n'est accessible que sur les dasri de synthèse."
     );
   }
 
   if (inputGrouping !== undefined) {
-    if (dbBsdasri.status !== BsdasriStatus.INITIAL) {
+    if (!isGroupingType) {
       throw new UserInputError(
-        "Les bordereaux associés à ce bsd ne sont plus modifiables"
+        "Le champ grouping n'est accessible que sur les dasri de groupement."
       );
     }
-
     if (!inputGrouping?.length) {
       throw new UserInputError(
         "Un bordereau de groupement doit avoir des bordereaux associés."
       );
     }
+    const groupingDiff =
+      inputGrouping.some(
+        id => !dbGrouping.some(({ id: dbId }) => dbId === id)
+      ) || inputGrouping.length !== dbGrouping.length;
```

Comment:
Je l'ai déjà dit mais tout le monde s'en carre, les conditions comme ça avec des ternaires, des if imbriqués et autres trucs complexes, je trouve ça difficile à lire. 

Perso je préfèrerais un petit:
```
const isGroupingDiff = isGroupingDifferent(inputGrouping, dbGrouping);
```

Et du code plus basique mais plus clair:
```
const isGroupingDifferent = (inputGrouping, dbGrouping) => {
  if(inputGrouping.length !== dbGrouping.length) return true;

  if(!inputGrouping.some(
        id => !dbGrouping.some(({ id: dbId }) => dbId === id)
      )){
    return false;
  }
};
```

En plus ta variable est un boolean si je ne m'abuse, elle devrait avoir un nom en `is` ou `has`.

Mais clairement je suis vieux jeu haha

Reply:
non je comprends parce que là je relis mon code et je comprends pas ce que ça fait donc c'est qu'il y a un soucis. Je vais modif